Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM python:2-alpine
+
+RUN apk add --no-cache --virtual .nacl_deps su-exec build-base libffi-dev zlib-dev openssl-dev libjpeg-turbo-dev linux-headers postgresql-dev
+
+COPY . /synapse
+
+# A wheel cache may be provided in ./cache for faster build
+RUN cd /synapse \
+ && pip install --upgrade pip setuptools psycopg2 \
+ && mkdir -p /synapse/cache \
+ && pip install -f /synapse/cache --upgrade --process-dependency-links . \
+ && mv /synapse/contrib/docker/* / \
```

Comment:
It seems strange to me to litter `/` with so many files.  Why not just keep them all there?

Reply:
I usually prefer to keep short and simple path for files in Docker images, like ``/start.py`` and ``/conf``. But you are right, other files should not be copied.